var/spack/repos/builtin/packages/readline/package.py

```diff
@@ -45,4 +45,6 @@ class Readline(AutotoolsPackage):
     patch('readline-6.3-upstream_fixes-1.patch', when='@6.3')
 
     def build(self, spec, prefix):
-        make('SHLIB_LIBS=-lncursesw')
+        options = ['SHLIB_LIBS={0}'.format(spec['ncurses'].libs.ld_flags)]
```

Comment:
unfortunately that's not doing what you want: 
```
options = ['SHLIB_LIBS=-L%s -lncursesw' % spec['ncurses'].prefix.lib]
```
The problems is that `libs` is an object `LibraryList`, with the default behaviour of trying to find a `lib<package>.dylib` in `<prefix>/lib`.

Whereas here we need `-lncursesw` (note `w` at the end), thus the default `LibraryList` won't contain what you need here.

One should eventually add non-default implementation of `libs` method to `ncurses`, but this is certainly outside of the scope of this PR.